pkg/utils/kube/resource_ops.go

```diff
@@ -229,7 +229,11 @@ func (k *kubectlResourceOperations) ApplyResource(ctx context.Context, obj *unst
 	span.SetBaggageItem("kind", obj.GetKind())
 	span.SetBaggageItem("name", obj.GetName())
 	defer span.Finish()
-	k.log.Info(fmt.Sprintf("Applying resource %s/%s in cluster: %s, namespace: %s", obj.GetKind(), obj.GetName(), k.config.Host, obj.GetNamespace()))
+	k.log.WithValues(
+		"dry-run", dryRunStrategy,
```

Comment:
Please implement the String function in the DryrunStrategy type so we can have a more meaningful logs.

Reply:
DryRunStrategy  is from kubectl library `k8s.io/kubectl/pkg/cmd/util` . So I get an error when adding String() fn
cannot define new methods on non-local type "k8s.io/kubectl/pkg/cmd/util".DryRunStrategy

Created an inline array lookup as below which is concise.
```
[...]string{"none", "client", "server"}[dryRunStrategy]
```
the messages now look like below for dry runs
```
time="2024-01-19T18:38:21Z" level=info msg="Applying resource Namespace/guestbook in cluster: [https://10.96.0.1:443](https://10.96.0.1/), namespace: " dry-run=client manager=argocd-controller serverSideApply=false serverSideDiff=false
```
and for the actual syncs, it looks like below:
```
time="2024-01-19T18:38:23Z" level=info msg="Applying resource Deployment/kustomize-guestbook-ui in cluster: [https://10.96.0.1:443](https://10.96.0.1/), namespace: guestbook" dry-run=none manager=argocd-controller serverSideApply=true serverSideDiff=false
```